lib/ansible/galaxy/collection.py

```diff
@@ -1419,10 +1447,17 @@ def _get_json_from_tar_file(b_path, filename):
 
 def _get_tar_file_hash(b_path, filename):
     with tarfile.open(b_path, mode='r') as collection_tar:
-        with _get_tar_file_member(collection_tar, filename) as tar_obj:
+        with _get_tar_file_member(collection_tar, filename) as (dummy, tar_obj):
             return _consume_file(tar_obj)
 
 
+def _is_child_path(path, parent_path):
+    """ Checks that path is a path within the parent_path specified. """
+    b_path = to_bytes(path, errors='surrogate_or_strict')
+    b_parent_path = to_bytes(parent_path, errors='surrogate_or_strict')
+    return b_path == b_parent_path or b_path.startswith(b_parent_path + to_bytes(os.path.sep))
```

Comment:
It might be better to use `os.path.relpath` and see whether it starts with `..`. Though for all "usual" cases this code should work, so let's keep it for now :)

Reply:
True, I would still need a check to see if the paths equal each other or at least equals `'.'`. I probably prefer that route so I'll do some more testing and update this code.